stock_picking_quick/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# © 2019 Today Akretion
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    move_ids = fields.One2many(
+        comodel_name='stock.move', inverse_name='product_id',
+        help='Technical: used to compute quantities to pick.',
+    )
+
+    @api.depends('move_ids')
+    def _compute_process_qty(self):
+        res = super(ProductProduct, self)._compute_process_qty()
+        if self.env.context['parent_model'] == 'stock.picking':
+            for product in self:
+                stock_moves = self.env['stock.move'].search([
+                    ('picking_id', '=', self.env.context.get('parent_id')),
+                    ('product_id', '=', product.id),
+                ])
+                for stock_move in stock_moves:
+                    product.qty_to_process += stock_move.product_qty
+        return res
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        picking = self.env['stock.picking'].browse(
+            self.env.context.get('parent_id'))
+        if self.env.context.get('in_current_picking') and picking:
```

Comment:
@PierrickBrun I'm really not a fan of parameters passed within the context.... Isn't it possible to define 'fake' computed fields with a search method so that they can be used in a normal domain?

Reply:
While thinking about it I realized the 'use_only_available_products' was the same as available_qty > 0. I'll look into it for the 'in_current_parent' filter